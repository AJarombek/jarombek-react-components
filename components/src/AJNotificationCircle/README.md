### Overview

Circle that shows a notification count.

**Use cases:**

```javascript
import React from 'react';
import { AJNotificationCircle } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJNotificationCircle count={5} />
```

### Files

| Filename                   | Description                                                                     |
|----------------------------|---------------------------------------------------------------------------------|
| `AJNotificationCircle.js`  | The reusable `AJNotificationCircle` component.                                  |
| `index.js`                 | Index file which exports the `AJNotificationCircle` component.                  |
| `styles.js`                | JSS styles for the `AJNotificationCircle` component.                            |
