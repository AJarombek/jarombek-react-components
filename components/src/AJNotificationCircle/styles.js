export default {
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: '50%',
    width: 20,
    height: 20,
    backgroundColor: '#333'
  },
  count: {
    fontSize: '12px',
    color: 'white',
    margin: 0,
    textAlign: 'center'
  }
};
