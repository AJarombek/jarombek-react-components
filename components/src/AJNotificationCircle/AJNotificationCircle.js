/**
 * Reusable circle which displays a notification count.
 * @author Andrew Jarombek
 * @since 8/14/2020
 */

import React from 'react';
import PropTypes from "prop-types";
import {createUseStyles} from 'react-jss';
import classnames from 'classnames';
import styles from './styles';

const useStyles = createUseStyles(styles);

const AJNotificationCircle = ({ count }) => {
  const classes = useStyles();

  return (
    <div className={classnames('aj-notification-circle', classes.container)}>
      <p className={classes.count}>{count}</p>
    </div>
  );
};

AJNotificationCircle.propTypes = {
  count: PropTypes.number
};

export default AJNotificationCircle;
