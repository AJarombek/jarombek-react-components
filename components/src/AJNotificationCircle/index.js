import AJNotificationCircle from './AJNotificationCircle';

export default AJNotificationCircle;
