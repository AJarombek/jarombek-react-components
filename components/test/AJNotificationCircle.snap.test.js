/**
 * Snapshot test for the {@link ../../components/AJNotificationCircle} component.
 * @author Andrew Jarombek
 * @since 8/15/2020
 */

import React from 'react';
import AJNotificationCircle from '../src/AJNotificationCircle/AJNotificationCircle';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AJNotificationCircle count={15} />).toJSON();
  expect(tree).toMatchSnapshot();
});
