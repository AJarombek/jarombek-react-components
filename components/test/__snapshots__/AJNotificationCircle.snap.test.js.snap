// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="aj-notification-circle container-0-2-1"
>
  <p
    className="count-0-2-2"
  >
    15
  </p>
</div>
`;
